### ISSUE TITLE
[FEATURE] Ability to add layers to a map in OpenSearch Dashboard

### ISSUE THREAD

**Issue by vamshin** [opensearch-project/maps#10]
**Is your feature request related to a problem?**
Yes [add layers to a map in OpenSearch Dashboard 1.2.0](https://discuss.opendistrocommunity.dev/t/how-to-add-layers-to-a-map-in-opensearch-dashboard-1-2-0/8654)

**What solution would you like?**
Ability to add layers to a map in OpenSearch Dashboard

**Comment by vamshin** [comment]
This issue should be transferred to Maps dashboard plugin repo once we create it.

**Comment by elfisher** [comment]
@vamshin are you thinking multiple layers from multiple indices? E.g., would this enable users to combine data across indices onto a single map visualization using multiple layers? That would enable users to correlate geographic data on the same visualization.

**Comment by vamshin** [comment]
@elfisher yes each layer can have different index/data source and they can all be shown on single map visualization. We can also control which layers should be visible at what zoom level  by associating zoom level to each of the layer we are adding. For example show only layer A, layerB for zoom 1, show layer X, layer Y for zoom 2 etc.
